sui/src/wallet_commands.rs

```diff
@@ -679,8 +692,8 @@ impl Display for WalletCommandResult {
             }
             WalletCommandResult::Objects(object_refs) => {
                 writeln!(writer, "Showing {} results.", object_refs.len())?;
-                for object_ref in object_refs {
-                    writeln!(writer, "{:?}", object_ref)?;
+                for (object_ref, lock_status) in object_refs {
+                    writeln!(writer, "{:?}, locked: {}", object_ref, lock_status)?;
```

Comment:
I am not sure if you want to expose this. No object should be locked in a normal flow (i.e. a typical user doesn't need to see this). It's purely implementation details that only us wants to see it.
So I would suggest make this only printed through an option

Reply:
I generally agree. My reason for exposing it was so one could see which objects not to use in TXes
I'm on the fence

Reply:
With both my fix on gas metering and your retry logic, I doubt this would happen (if it does we should fix it asap anyway instead of giving people a way to get around it).

Reply:
fair enough